DbmProgram/Program.cs

```diff
@@ -190,24 +127,73 @@ public static UnityContainer ConfigureContainer()
 
     class Program
     {
-        private static void RunOptionsAndReturnExitCode(Options opts)
+        private static void RunOptionsAndReturnExitCode(Options opts, string[] args)
         {
             UnityContainer ScriptContainer = Factory.ConfigureContainer();
+            ScriptContainer.RegisterInstance(opts);
             ScriptController scriptController = ScriptContainer.Resolve<ScriptController>();
-            scriptController.RunScript(opts);
+            ConfigService configService = ConfigService.getConfig();
+            configService.SetUp(opts);
+            string rootPath = opts.RootPath;
+            // validaten parameters
+            if (!(opts.IsValidConn() && opts.IsValidPath() && opts.IsValidSubsituteList()))
+            {
+                scriptController.ExitFailureProgram($"Overall Status: failure\n{(opts.IsValidConn() ? null : "<ConnString> or <Dbname>")} {(opts.IsValidPath() ? null : "<RootPath>")} {(opts.IsValidSubsituteList() ? null : "<SubstituteList>")} Argument is not valid", 0);
+            }
+            if (opts.IsSnapshot)
+                scriptController.CreateSnapshotDatabase(opts.ConnString);
+            if (opts.IsRunningUI) {
+                try
```

Comment:
refactor to method - better yet separate class. Your code should read like a table of contents. Here you have code at different levels of abstraction. Makes it difficult to follow